custom/search.js

```diff
@@ -55,20 +55,20 @@ async function fullSearch({drive, query, folderIds, results = [], nextPageToken:
 }
 
 // Grab all folders in directory to search through in shared drive
-async function getAllFolders({nextPageToken: pageToken, drive, parentIds = [driveId], foldersSoFar = []} = {}) {
+async function getAllFolders({nextPageToken, drive, parentIds = [driveId], foldersSoFar = []} = {}) {
   const options = {
     ...list.commonListOptions.folder,
     q: `(${parentIds.map((id) => `'${id}' in parents`).join(' or ')}) AND mimeType = 'application/vnd.google-apps.folder'`,
     fields: 'files(id,name,mimeType,parents)'
   }
 
-  if (pageToken) {
-    options.pageToken = pageToken
+  if (nextPageToken) {
+    options.nextPageToken = nextPageToken
   }
 
   const {data} = await drive.files.list(options)
 
-  const {files, nextPageToken} = data
+  const {files} = data
   const combined = foldersSoFar.concat(files)
 
   if (nextPageToken) {
```

Comment:
this looks like you might trigger infinite recursion now --- if you pass in `nextPageToken` to `getAllFolders`, then continue calling `getAllFolders` again?

Reply:
I can look into it, but none of the logic really changed here

Reply:
wdym "none of the logic really changed here"?

line 74 checks if there's a `nextPageToken` and then calls again to get the next page of data. but if we're no longer pulling `nextPageToken` from the response, then how does it work?